src/coreclr/nativeaot/Runtime/startup.cpp

```diff
@@ -508,20 +511,6 @@ extern "C" bool RhInitialize()
     return true;
 }
 
-//
-// Currently called only from a managed executable once Main returns, this routine does whatever is needed to
-// cleanup managed state before exiting. There's not a lot here at the moment since we're always about to let
-// the OS tear the process down anyway.
-//
-// @TODO: Eventually we'll probably have a hosting API and explicit shutdown request. When that happens we'll
-// something more sophisticated here since we won't be able to rely on the OS cleaning up after us.
-//
-COOP_PINVOKE_HELPER(void, RhpShutdown, ())
-{
-    // Indicate that runtime shutdown is complete and that the caller is about to start shutting down the entire process.
-    g_threadPerformingShutdown = ThreadStore::RawGetCurrentThread();
-}
-
 #ifdef _WIN32
 EXTERN_C UInt32_BOOL WINAPI RtuDllMain(HANDLE hPalInstance, uint32_t dwReason, void* pvReserved)
```

Comment:
I do not see any place where this is used. Delete it?

Reply:
I assumed `RtuDllMain` is used somehow when a NativeAOT dll is created. It is not used as-is.

Reply:
I do not think it is used anywhere. It looks like a left-over from redhawk.

Reply:
It looks like, at least in a case of .dll publishing, one would need to hookup `InitDLL`/`UninitDLL`, and on windows `OnProcessExit`.  

Is that handled somehow or dynamic NativeAOT libraries are not supported?

Reply:
I see we have a [static lib sample](https://github.com/dotnet/samples/tree/main/core/nativeaot/NativeLibrary), but no dlls.

Reply:
> I see we have a [static lib sample](https://github.com/dotnet/samples/tree/main/core/nativeaot/NativeLibrary), but no dlls.

The DLL sample is in the same location: https://github.com/dotnet/samples/tree/main/core/nativeaot/NativeLibrary#building-shared-libraries

Building DLLs is the only officially documented thing. Static lib is just the sample: https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/#build-native-libraries

Reply:
Hmm I missed that, how does the .dll initialize the runtime? Calls `RtuDllMain` somehow, or calls `RhInitialize` on the first reverse pinvoke?
The latter should work (and perhaps more reliably) with this change. 
Do we have a test?

Reply:
I think we initialize on first reverse p/invoke. The test is here: https://github.com/dotnet/runtime/tree/main/src/tests/nativeaot/SmokeTests/SharedLibrary